src/test.rs

```diff
@@ -370,6 +371,73 @@ fn parse_test_case(current_section: &mut String, lines: &mut FileLines)
     }
 }
 
+/// A container that stores a byte sequence preceded by a prefix of zero
+/// or more dummy bytes.  This is useful for test cases that need to exercise
+/// vectorized implementations with non-aligned inputs.
+pub struct PrefixedByteVec {
+    prefix_len: usize,
+    vec: Vec<u8>,
+}
+
+impl PrefixedByteVec {
+    /// Create an empty PrefixedByteVec that can hold `size` bytes of
+    /// prefix+data.
+    pub fn with_capacity(size: usize) -> PrefixedByteVec {
+        PrefixedByteVec {
+            prefix_len: 0,
+            vec: Vec::with_capacity(size),
+        }
+    }
+
+    /// Create an PrefixedByteVec containing a copy of the provided slice,
+    /// preceded by `prefix_len` dummy bytes.
+    pub fn from(prefix_len: usize, data: &[u8]) -> PrefixedByteVec {
+        let mut v = PrefixedByteVec {
+            prefix_len: prefix_len,
+            vec: Vec::with_capacity(prefix_len + data.len()),
+        };
+        v.replace_from(prefix_len, data);
+        v
+    }
+
+    /// Replace a PrefixedByteVec's contents with a copy of the provided
+    /// slice, preceded by `prefix_len` dummy bytes.
+    pub fn replace_from(&mut self, prefix_len: usize, data: &[u8]) {
+        self.vec.clear();
+        for _ in 0..prefix_len {
+            self.vec.push(123);
```

Comment:
Here we may allocate or re-allocate the storage one or more times. Potentially, these allocations could have differing alignments, such that when `alignment + prefix_len` results in duplicates, and such that some alignments wouldn't be tested.